hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -3421,6 +3421,17 @@ private void autoAdjustConfigsForConcurrencyMode(boolean isLockProviderPropertyS
             HoodieLockConfig.LOCK_PROVIDER_CLASS_NAME.key(), InProcessLockProvider.class.getName()));
       }
 
+      // for a single writer scenario, with all table services inline, lets set InProcessLockProvider
+      if (writeConfig.getWriteConcurrencyMode() == WriteConcurrencyMode.SINGLE_WRITER && !writeConfig.areAnyTableServicesAsync()) {
+        if (writeConfig.getLockProviderClass() != null) {
+          // add logs only when explicitly overridden by the user.
+          LOG.warn(String.format("For a single writer mode, overriding lock provider class (%s) to %s. So, user configured lock provider %s may not take effect",
```

Comment:
Should it be a warning? Will come up a lot.

Reply:
yes, we are printing this only if user explicitly set it to some lock provider. for out of the box, it should not print.

Reply:
actually pushed an update to ensure we only log if user configures some LP which is not InProcessLP